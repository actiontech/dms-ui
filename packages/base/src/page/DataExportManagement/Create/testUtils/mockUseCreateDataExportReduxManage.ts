import { CreateDataExportPageEnum } from '../../../../store/dataExport';
import { ExportMethodEnum } from '../components/CreateTask/ExportMethodForm/index.enum';
import * as useCreateDataExportReduxManage from '../hooks/index.redux';

export const mockCreateDataExportRedux = {
  updateDataExportInfoOpen: false,
  formValues: {
    baseValues: { workflow_subject: 'test', desc: 'desc' },
    sourceValues: { dbService: '121' },
    methodValues: { sql: 'select 1;', exportMethod: ExportMethodEnum.sql }
  },
  pageState: CreateDataExportPageEnum.CREATE_TASK,
  auditLoading: false,
  submitLoading: false,
  taskIDs: ['11111'],
  workflowID: '123',
  updateFormValues: jest.fn(),
  updatePageState: jest.fn(),
  updateAuditLoading: jest.fn(),
  updateSubmitLoading: jest.fn(),
  initModalStatus: jest.fn(),
  updateModalStatus: jest.fn(),
  updateTaskIDs: jest.fn(),
  updateWorkflowID: jest.fn(),
  clearAllState: jest.fn()
};

export const mockUseCreateDataExportReduxManage = (
  data?: Partial<typeof mockCreateDataExportRedux>
) => {
  const spy = jest.spyOn(useCreateDataExportReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockCreateDataExportRedux, ...data }));
  return spy;
};
