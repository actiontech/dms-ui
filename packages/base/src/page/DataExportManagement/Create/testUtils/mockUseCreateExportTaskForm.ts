import * as useCreateExportTaskForm from '../hooks/useCreateExportTaskForm';

export const mockUseCreateExportTaskFormReturn = {
  baseForm: null as any,
  sourceForm: null as any,
  methodForm: null as any,
  auditLoading: false,
  formatSQLAction: jest.fn(),
  auditAction: jest.fn(),
  resetAllForms: jest.fn()
};

export const mockUseCreateExportTaskForm = (
  data?: Partial<typeof mockUseCreateExportTaskFormReturn>
) => {
  const spy = jest.spyOn(useCreateExportTaskForm, 'default');
  spy.mockImplementation(() => ({
    ...mockUseCreateExportTaskFormReturn,
    ...data
  }));
  return spy;
};
