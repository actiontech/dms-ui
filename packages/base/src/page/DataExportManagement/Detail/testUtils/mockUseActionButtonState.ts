import * as useActionButtonState from '../components/PageHeaderAction/useActionButtonState';

export const mockActionButtonStateData = {
  closeWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  approveWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  rejectWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  executeExportButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  }
};

export const mockUseActionButtonState = (
  data?: Partial<typeof mockActionButtonStateData>
) => {
  const spy = jest.spyOn(useActionButtonState, 'default');
  spy.mockImplementation(() => ({ ...mockActionButtonStateData, ...data }));
  return spy;
};
