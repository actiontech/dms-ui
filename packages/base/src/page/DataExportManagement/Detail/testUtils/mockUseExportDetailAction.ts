import * as useExportDetailAction from '../hooks/useExportDetailAction';

export const mockExportDetailActionData = {
  refreshWorkflow: jest.fn(),
  closeWorkflowLoading: false,
  closeWorkflow: jest.fn(),
  approveWorkflowLoading: false,
  approveWorkflow: jest.fn(),
  executeExportLoading: false,
  executeExport: jest.fn()
};

export const mockUseExportDetailAction = (
  data?: Partial<typeof mockExportDetailActionData>
) => {
  const spy = jest.spyOn(useExportDetailAction, 'default');
  spy.mockImplementation(() => ({ ...mockExportDetailActionData, ...data }));
  return spy;
};
