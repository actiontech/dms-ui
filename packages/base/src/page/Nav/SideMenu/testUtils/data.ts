export const mockRecentlyOpenedProjectsData = {
  recentlyProjects: [
    {
      project_id: '1',
      project_name: 'test_project'
    }
  ],
  updateRecentlyProject: jest.fn(),
  currentProjectID: '0'
};
