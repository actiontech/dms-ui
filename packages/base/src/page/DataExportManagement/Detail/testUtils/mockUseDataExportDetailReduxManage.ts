import { GetDataExportTaskStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import * as useDataExportDetailReduxManage from '../hooks/index.redux';
import { GetDataExportWorkflowResponseData } from '../../../../testUtils/mockApi/dataExport/data';

export const mockDataExportDetailRedux = {
  workflowStepOpen: false,
  workflowRejectOpen: false,
  workflowInfo: GetDataExportWorkflowResponseData,
  taskInfos: [
    {
      task_uid: '1752172791873933312',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.finish,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933312.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933313',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.exporting,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933313.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933314',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.failed,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933314.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    }
  ],
  curTaskID: '1752172791873933312',
  taskStatusNumber: {
    success: 1,
    failed: 0,
    exporting: 0
  },
  canRejectWorkflow: false,
  updateWorkflowStepOpen: jest.fn(),
  updateWorkflowRejectOpen: jest.fn(),
  updateWorkflowInfo: jest.fn(),
  updateTaskInfos: jest.fn(),
  updateCurTaskID: jest.fn(),
  updateTaskStatusNumber: jest.fn(),
  updateCanRejectWorkflow: jest.fn(),
  clearAllDetailState: jest.fn()
};

export const mockUseDataExportDetailReduxManage = (
  data?: Partial<typeof mockDataExportDetailRedux>
) => {
  const spy = jest.spyOn(useDataExportDetailReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockDataExportDetailRedux, ...data }));
  return spy;
};
