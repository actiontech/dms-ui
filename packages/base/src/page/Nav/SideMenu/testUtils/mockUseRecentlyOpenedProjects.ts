import * as useRecentlyOpenedProjects from '../useRecentlyOpenedProjects';
import { mockRecentlyOpenedProjectsData } from './data';

export const mockUseRecentlyOpenedProjects = (
  data?: Partial<typeof mockRecentlyOpenedProjectsData>
) => {
  const spy = jest.spyOn(useRecentlyOpenedProjects, 'default');
  spy.mockImplementation(() => ({
    ...mockRecentlyOpenedProjectsData,
    ...data
  }));
  return spy;
};
