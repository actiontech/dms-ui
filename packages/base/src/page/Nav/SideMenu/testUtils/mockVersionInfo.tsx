import * as useVersionInfo from '../UserMenu/hooks/useVersionInfo';
import { mockVersionInfoData } from './data';

export const mockVersionInfo = (data?: Partial<typeof mockVersionInfoData>) => {
  const spy = jest.spyOn(useVersionInfo, 'default');
  spy.mockImplementation(() => ({
    ...mockVersionInfoData,
    ...data
  }));
};
