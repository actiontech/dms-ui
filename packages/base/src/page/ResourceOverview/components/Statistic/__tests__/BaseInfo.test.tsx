import { superRender } from '../../../../../testUtils/customRender';
import ResourceOverviewBaseInfo from '../BaseInfo';
import resourceOverview from '../../../../../testUtils/mockApi/resourceOverview';
import { cleanup, act } from '@testing-library/react';
import EmitterKey from '../../../../../data/EmitterKey';
import EventEmitter from '../../../../../utils/EventEmitter';

describe('base/page/ResourceOverview/Statictis/BaseInfo', () => {
  let getResourceOverviewStatisticsV1Spy: jest.SpyInstance;
  beforeEach(() => {
    getResourceOverviewStatisticsV1Spy =
      resourceOverview.getResourceOverviewStatisticsV1();
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
    cleanup();
  });

  it('render init snap', async () => {
    const { container } = superRender(<ResourceOverviewBaseInfo />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(container).toMatchSnapshot();
    expect(getResourceOverviewStatisticsV1Spy).toHaveBeenCalledTimes(1);
  });

  it('should refresh statictis data when emit Refresh_Resource_Overview_Page event', async () => {
    superRender(<ResourceOverviewBaseInfo />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getResourceOverviewStatisticsV1Spy).toHaveBeenCalledTimes(1);
    await act(async () => {
      EventEmitter.emit(EmitterKey.Refresh_Resource_Overview_Page);
      await jest.advanceTimersByTime(300);
    });
    expect(getResourceOverviewStatisticsV1Spy).toHaveBeenCalledTimes(2);
  });
});
