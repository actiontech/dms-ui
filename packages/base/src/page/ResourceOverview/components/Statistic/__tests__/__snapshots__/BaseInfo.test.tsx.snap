// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/page/ResourceOverview/Statictis/BaseInfo render init snap 1`] = `
<div>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-space ant-space-vertical statistic-card-wrapper full-width-element"
      >
        <div
          class="ant-space-item"
          style="margin-bottom: 24px;"
        >
          <div
            class="ant-card ant-card-bordered ant-card-hoverable statistic-card"
          >
            <div
              class="ant-card-body"
            >
              <div
                class="statistic-card-content"
              >
                <div
                  class="statistic-card-icon"
                >
                  <svg
                    color="#15C7D4"
                    height="24"
                    viewBox="0 0 18 18"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8.25 5.25V3A.75.75 0 0 1 9 2.25h6.75a.75.75 0 0 1 .75.75v12a.75.75 0 0 1-.75.75H2.25A.75.75 0 0 1 1.5 15V6a.75.75 0 0 1 .75-.75zM3.75 12v1.5H7.5V12zm6.75 0v1.5h3.75V12zm0-2.25v1.5h3.75v-1.5zm0-2.25V9h3.75V7.5zM3.75 9.75v1.5H7.5v-1.5z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="ant-statistic"
                >
                  <div
                    class="ant-statistic-title"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        业务总数
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-statistic-content"
                  >
                    <span
                      class="ant-statistic-content-value"
                    >
                      <span
                        class="ant-statistic-content-value-int"
                      >
                        1
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
          style="margin-bottom: 24px;"
        >
          <div
            class="ant-card ant-card-bordered ant-card-hoverable statistic-card"
          >
            <div
              class="ant-card-body"
            >
              <div
                class="statistic-card-content"
              >
                <div
                  class="statistic-card-icon"
                >
                  <svg
                    height="24"
                    viewBox="0 0 18 18"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.25 2.25h7.037a.75.75 0 0 1 .67.415L10.5 3.75H15a.75.75 0 0 1 .75.75v8.25a.75.75 0 0 1-.75.75h-4.786a.75.75 0 0 1-.671-.415L9 12H3.75v4.5h-1.5z"
                      fill="#EBAD1C"
                    />
                  </svg>
                </div>
                <div
                  class="ant-statistic"
                >
                  <div
                    class="ant-statistic-title"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        项目总数
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-statistic-content"
                  >
                    <span
                      class="ant-statistic-content-value"
                    >
                      <span
                        class="ant-statistic-content-value-int"
                      >
                        3
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
        >
          <div
            class="ant-card ant-card-bordered ant-card-hoverable statistic-card"
          >
            <div
              class="ant-card-body"
            >
              <div
                class="statistic-card-content"
              >
                <div
                  class="statistic-card-icon"
                >
                  <svg
                    color="#4583FF"
                    height="24"
                    viewBox="0 0 18 18"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.75 7.125v2.25c0 1.864-3.023 3.375-6.75 3.375s-6.75-1.511-6.75-3.375v-2.25C2.25 8.989 5.273 10.5 9 10.5s6.75-1.511 6.75-3.375m-13.5 3.75c0 1.864 3.023 3.375 6.75 3.375s6.75-1.511 6.75-3.375v2.25c0 1.864-3.023 3.375-6.75 3.375s-6.75-1.511-6.75-3.375zM9 9C5.273 9 2.25 7.489 2.25 5.625S5.273 2.25 9 2.25s6.75 1.511 6.75 3.375S12.727 9 9 9"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="ant-statistic"
                >
                  <div
                    class="ant-statistic-title"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        数据源总数
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-statistic-content"
                  >
                    <span
                      class="ant-statistic-content-value"
                    >
                      <span
                        class="ant-statistic-content-value-int"
                      >
                        2
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
